lib/validate_url.rb

```diff
@@ -53,18 +54,25 @@ def validate_url(record, attribute, value, schemes)
         uri = URI.parse(value)
         host = uri && uri.host
         scheme = uri && uri.scheme
+        url = value.sub("#{scheme}://", '')
 
         valid_raw_url = scheme && value =~ /\A#{URI::regexp([scheme])}\z/
         valid_scheme = host && scheme && schemes.include?(scheme)
         valid_no_local = !options.fetch(:no_local) || (host && host.include?('.'))
         valid_suffix = !options.fetch(:public_suffix) || (host && PublicSuffix.valid?(host, :default_rule => nil))
+        valid_uniq = options.fetch(:uniq_url) ? url_uniq?(record, attribute, url) : true
 
-        unless valid_raw_url && valid_scheme && valid_no_local && valid_suffix
+        unless valid_raw_url && valid_scheme && valid_no_local && valid_suffix && valid_uniq
           record.errors.add(attribute, options.fetch(:message), value: value)
         end
       rescue URI::InvalidURIError
         record.errors.add(attribute, :url, **filtered_options(value))
       end
+
+      # Validates the URL is unique without the schema
+      def url_uniq?(record, attribute, url)
+        record.class.where("#{attribute} LIKE ?", "%#{url}").where.not(id: record.id).size.zero?
```

Comment:
Is there a better way to interpolate this? I tried many things...

Reply:
nah I wouldn't worry, it looks fine, does what we need :)

Reply:
Alternative could be interpolation with named params, but I think it's more wordy for the sake of it... yours is clearer.
```
record.class.where(":attribute LIKE :url", attribute: attribute, url: "%#{url}")
```